src/app/resolver/feature-flag.resolver.ts

```diff
@@ -0,0 +1,147 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  Resolve,
+  Router,
+  RouterStateSnapshot
+} from '@angular/router';
+import { Logger } from 'ngx-base';
+import { UserService } from 'ngx-login-client';
+import { Observable } from 'rxjs';
+
+import { Feature } from '../models/feature';
+import { FeatureFlagConfig } from '../models/feature-flag-config';
+import { FeatureTogglesService } from '../service/feature-toggles.service';
+
+
+enum FeatureLevel {
+  internal = 'internal',
+  released = 'released',
+  notApplicable = 'notApplicable', // non redhat user trying to access internal feature
+  systemError = 'systemError', // f8-toggles-service is down, this features is disabled by PM for all level
+  experimental = 'experimental',
+  beta = 'beta'
+}
+@Injectable()
+export class FeatureFlagResolver implements Resolve<FeatureFlagConfig> {
+  userLevel: string = 'released';
+  constructor(private logger: Logger,
+              private toggleService: FeatureTogglesService,
+              private router: Router,
+              private userService: UserService) {
+    if (this.userService.currentLoggedInUser && this.userService.currentLoggedInUser.attributes) {
+      this.userLevel = (this.userService.currentLoggedInUser.attributes as any).featureLevel;
+    }
+  }
+   buildFeaturePerLevelView(features: Feature[], userLevel: string): FeatureFlagConfig {
```

Comment:
It looks like something weird was happening with the indentation in this file, even before it was moved across repos. Could you reindent it please since we're already here?